cpp/mcap/include/mcap/reader.hpp

```diff
@@ -131,7 +132,7 @@ class ICompressedReader : public IReadable {
  * @brief A "null" compressed reader that directly passes through uncompressed
  * data. No internal buffers are allocated.
  */
-class BufferReader final : public ICompressedReader {
+class MCAP_PUBLIC BufferReader final : public ICompressedReader {
```

Comment:
does it make sense that BufferReader is public but ICompressedReader is not? I get a warning about this here: https://github.com/foxglove/mcap/actions/runs/3415978239/jobs/5685665885#step:9:153

Reply:
It does not make sense, no. I've fixed it